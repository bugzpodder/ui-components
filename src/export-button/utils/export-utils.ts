import {
  ExportableColumn,
  ReportOptions,
  toDelimitedReport,
} from "@grailbio/lib";
import { saveAs } from "file-saver";

export const generateReport = (
  fileName: string,
  columns: ExportableColumn<any>[],
  data: Record<string, any>[],
  options: ReportOptions = {},
): void => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
