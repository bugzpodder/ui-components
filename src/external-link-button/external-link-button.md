###### Optional

- Any props for a Material UI Button will take, such as `className`

### Example

```js
const { ExternalLinkButton } = require("./external-link-button");

<ExternalLinkButton variant="raised" color="primary" href="https://www.grail.com">
	GRAIL Link in new tab
</ExternalLinkButton>;
```
