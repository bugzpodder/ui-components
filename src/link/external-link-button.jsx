// @flow
import Button from "@material-ui/core/Button";
import React, { type ElementConfig, type Node } from "react";

type Props = {
  /** The children to display in the body of the button */
  children?: Node,
} & ElementConfig<typeof Button>;

/**
 *  ExternalLinkButton opens in a new tab, while ensuring that the link protects from:
 * https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/.
 * Similar to ExternalLink but it is represented in a button.
 */
export const ExternalLinkButton = (props: Props) => {
  const { children, ...buttonProps } = props;
  return (
    <Button
      data-testid="external-link-button"
      {...buttonProps}
      target="_blank"
      rel="noopener noreferrer"
    >
      {children}
    </Button>
  );
};
