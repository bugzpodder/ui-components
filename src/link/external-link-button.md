###### Optional

- Any props for a Material UI Button will take, such as `className`

### Example

```js
import { ExampleWrapper } from "../test-utils";
import { ExternalLinkButton } from "./";
<ExampleWrapper>
  <ExternalLinkButton variant="contained" color="primary" href="https://www.grail.com">
    GRAIL Link in new tab
  </ExternalLinkButton>
</ExampleWrapper>;
```
