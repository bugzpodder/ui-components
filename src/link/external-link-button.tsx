import Button, { ButtonProps } from "@material-ui/core/Button";

import React from "react";
import classNames from "classnames";
import styles from "./link-button.module.scss";

/**
 *  ExternalLinkButton opens in a new tab, while ensuring that the link protects from:
 * https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/.
 * Similar to ExternalLink but it is represented in a button.
 */
export const ExternalLinkButton: React.FC<ButtonProps> = props => {
  const { children, className, ...buttonProps } = props;
  const button = (
    // @ts-ignore property target does not exist.
    <Button
      data-testid="external-link-button"
      {...buttonProps}
      role="button"
      className={classNames(className, styles.link)}
      target="_blank"
      rel="noopener noreferrer"
    >
      {children}
    </Button>
  );
  return <div className={styles.button}>{button}</div>;
};
