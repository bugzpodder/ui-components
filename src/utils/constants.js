// @flow
export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  value: suggestion.label,
  label: suggestion.label,
}));
