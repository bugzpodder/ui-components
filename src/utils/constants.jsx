// @flow
import React from "react";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const SERGEY =
  "https://media.licdn.com/dms/image/C5603AQGlBIPst1V4Ew/profile-displayphoto-shrink_800_800/0?e=1535587200&v=beta&t=2V2knyEhbTrhRWzgydsXLZqiX8gnoD1poj8jGeFpA0w";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];
