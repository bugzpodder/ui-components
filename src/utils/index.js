// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
