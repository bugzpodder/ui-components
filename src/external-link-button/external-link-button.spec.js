// @flow
import React from "react";
import { ExternalLinkButton } from "./external-link-button";
import { TestWrapper } from "../utils";
import { cleanup, render } from "react-testing-library";

afterEach(cleanup);

test("render ExternalLink", () => {
  const { container } = render(
    <TestWrapper>
      <ExternalLinkButton
        href="https://www.grail.com"
        className="test"
      >
        GRAIL
      </ExternalLinkButton>
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});
