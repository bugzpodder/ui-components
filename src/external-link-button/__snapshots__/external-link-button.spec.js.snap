// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExternalLink 1`] = `
<div>
  <a
    class="MuiButtonBase-root MuiButton-root MuiButton-flat test"
    href="https://www.grail.com"
    rel="noopener noreferrer"
    role="button"
    tabindex="0"
    target="_blank"
  >
    <span
      class="MuiButton-label"
    >
      GRAIL
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;
